contrib/storage-kafka/src/main/java/org/apache/drill/exec/store/kafka/decoders/AvroMessageReader.java

```diff
@@ -85,10 +90,19 @@ public void readMessage(ConsumerRecord<?, ?> record) {
     writeValue(rowWriter, MetaDataField.KAFKA_PARTITION_ID, record.partition());
     writeValue(rowWriter, MetaDataField.KAFKA_OFFSET, record.offset());
     writeValue(rowWriter, MetaDataField.KAFKA_TIMESTAMP, record.timestamp());
-    writeValue(rowWriter, MetaDataField.KAFKA_MSG_KEY, record.key() != null ? record.key().toString() : null);
+    writeValue(rowWriter, MetaDataField.KAFKA_MSG_KEY, record.key() != null ? getKeyValue((byte[]) record.key()) : null);
     rowWriter.save();
   }
 
+  private Object getKeyValue(byte[] keyValue) {
+    if (keyConvert) {
+      GenericRecord genericRecord = (GenericRecord) deserializer.deserialize(null, keyValue);
+      return genericRecord.toString();
```

Comment:
No need to cast it to `GenericRecord`:
```suggestion
      deserializer.deserialize(null, keyValue).toString();
```